Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Last-Update: 2012-09-08
Forwarded: 2012-09-08
Bug-abiword: http://bugzilla.abisource.com/show_bug.cgi?id=13400
Description: fixes two problems with collab-backend if built using "--with-gtk2":
 Because [ "$TOOLKIT" = "gtk2" ] condition is always false,
 collab-backend-xmpp always requests gtk3. Therefore build using
  --with-gtk2 --enable-collab-backend-xmpp
 FTBFS without gtk3 development files installed (i.e. "libgtk-3-dev" package).
 .
 In presence of libgtk-3-dev --enable-collab-backend-xmpp always links to gtk3
 even if built using "--with-gtk2" which completely breaks collaborate menu.

--- a/plugins/collab/plugin.m4
+++ b/plugins/collab/plugin.m4
@@ -1,7 +1,7 @@
 
 collab_req="libgsf-1 >= 1.12 libxml-2.0 >= 2.4.0"
 collab_telepathy_req="dbus-glib-1 >= 0.70 telepathy-glib >= 0.14.5"
-if test "$TOOLKIT" = "gtk2"; then
+if test "$TOOLKIT_IS_GTK2" = "yes"; then
 collab_xmpp_req="loudmouth-1.0 >= 1.3.2 gtk+-2.0"
 else
 collab_xmpp_req="loudmouth-1.0 >= 1.3.2 gtk+-3.0"
--- a/plugins/aiksaurus/plugin.m4
+++ b/plugins/aiksaurus/plugin.m4
@@ -7,7 +7,7 @@
 
 PKG_CHECK_EXISTS([ $aiksaurus_pkgs ], 
 [
-	if test "$TOOLKIT" = "gtk2"; then
+	if test "$TOOLKIT_IS_GTK2" = "yes"; then
 	PKG_CHECK_EXISTS([ $aiksaurus_gtk_pkgs ], 
 	[
 		aiksaurus_deps="yes"
@@ -32,7 +32,7 @@
 
 PKG_CHECK_MODULES(AIKSAURUS,[ $aiksaurus_pkgs ])
 
-if test "$TOOLKIT" = "gtk2"; then
+if test "$TOOLKIT_IS_GTK2" = "yes"; then
 	PKG_CHECK_MODULES(AIKSAURUS_GTK,[ $aiksaurus_gtk_pkgs ])
 	AIKSAURUS_CFLAGS="$AIKSAURUS_CFLAGS $AIKSAURUS_GTK_CFLAGS"
 	AIKSAURUS_LIBS="$AIKSAURUS_LIBS $AIKSAURUS_GTK_LIBS"
